src/main/java/com/library/model/mapper/BookMapper.java

```diff
@@ -1,20 +1,30 @@
 package com.library.model.mapper;
 
 import com.library.model.dto.BookDto;
-import com.library.model.entity.Book;
-import lombok.RequiredArgsConstructor;
+import com.library.model.entity.postgres.Book;
 import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Component;
 
 /**
  * Book converter based on ModelMapper to map Book in BookDto and vice versa
  */
 @Component
-@RequiredArgsConstructor
 public class BookMapper {
 
     private final ModelMapper modelMapper;
 
+    public BookMapper(ModelMapper modelMapper){
+        this.modelMapper = modelMapper;
+        modelMapper.createTypeMap(BookDto.class, com.library.model.entity.mongo.Book.class)
+                .addMappings(mapping -> mapping.skip(com.library.model.entity.mongo.Book::setId))
+                .<Integer>addMapping(BookDto::getStack, ((destination, value) -> destination.getBookshelf().setStack(value)))
```

Comment:
Info about library's book shelf in book storage. For what is it?

Reply:
So that to put it on the same shelf when return to library

Reply:
We did not discuss fuctionality so let it be in this way.

Why info about shelf in document became separate bean?

Reply:
Because same Book(entity) fields doesn't have relation with BookStored fields, and your task "...Например, что делать с "адресом" книги на полке при перемещениях книги в/из хранилища..." was solved in this way.

Reply:
Ок.